pkg/expect/expect.go

```diff
@@ -223,14 +229,23 @@ func (ep *ExpectProcess) ExpectFunc(ctx context.Context, f func(string) bool) (s
 }
 
 // ExpectWithContext returns the first line containing the given string.
-func (ep *ExpectProcess) ExpectWithContext(ctx context.Context, s string) (string, error) {
-	return ep.ExpectFunc(ctx, func(txt string) bool { return strings.Contains(txt, s) })
+func (ep *ExpectProcess) ExpectWithContext(ctx context.Context, s ExpectedResponse) (string, error) {
+	return ep.ExpectFunc(ctx, func(txt string) bool {
+		if s.IsRegularExpr {
```

Comment:
I think we can Compile out of closure and return error as soon as possible. And then we don't need to Compile it all the time.

Reply:
Makes sense & updated & thx